network/discovery/src/file.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::DiscoveryChangeListenerTrait;
+use async_trait::async_trait;
+use diem_config::{config::PeerSet, network_id::NetworkContext};
+use diem_logger::prelude::*;
+use diem_time_service::{Sleep, TimeService, TimeServiceTrait};
+use futures::{Future, Stream};
+use network::{connectivity_manager::DiscoverySource, logging::NetworkSchema, ConnectivityRequest};
+use std::{
+    path::{Path, PathBuf},
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+    time::Duration,
+};
+
+/// Watcher for a file containing discovered peers at a given interval
+pub struct FileChangeListener {
+    network_context: Arc<NetworkContext>,
+    file_path: PathBuf,
+    update_channel: channel::Sender<ConnectivityRequest>,
+    time_service: TimeService,
+    interval_duration: Duration,
+    delay: Option<Pin<Box<Sleep>>>,
```

Comment:
should just be a `time_service::Interval` here

Reply:
Interval pls : p

Reply:
I can do this, but to do it properly `NetworkBuilder` needs to be reworked so executor is passed in at an earlier time instead of the wonky split system it is now.